common/etc/nginx/templates/gateway/cors.conf.template

```diff
@@ -11,6 +12,18 @@ if ($request_cors = "OPTIONS_1") {
     # Tell client that this pre-flight info is valid for 20 days
     #
     add_header 'Access-Control-Max-Age' 1728000;
+    #
+    # Allow/deny Private Network Access CORS requests.
+    # https://developer.chrome.com/blog/private-network-access-preflight/
+    #
+    if ($http_access_control_request_private_network = "true") {
+        if ($cors_private_network = "true") {
+            add_header 'Access-Control-Allow-Private-Network' 'true';
+        }
+        if ($cors_private_network = "false") {
+            add_header 'Access-Control-Allow-Private-Network' 'false';
+        }
```

Comment:
Unfortunately this code is not going to work since the `if` directive can't be used in the `if` context (no nested `if`s).

If we decide that it's OK not to make the `CORS_ALLOW_PRIVATE_NETWORK_ACCESS` configuration object dependent on `CORS_ENABLED` and trust users to enable both themselves, then we can write something like:

```
set $cors_private_network "${CORS_ALLOW_PRIVATE_NETWORK_ACCESS}";

if ($cors_private_network = 'true') {
    set $allow_cors_private_network $http_access_control_request_private_network;
}

if ($cors_private_network = 'false') {
    set $allow_cors_private_network 'false';
}

if ($request_cors = "OPTIONS_1") {
    # ...
    # Allow/deny Private Network Access CORS requests.
    # https://developer.chrome.com/blog/private-network-access-preflight/
    #
    add_header 'Access-Control-Allow-Private-Network' $allow_cors_private_network;

```

This would result in the following behavior:

- If `CORS_ALLOW_PRIVATE_NETWORK_ACCESS` is set to `"true"`, the `Access-Control-Allow-Private-Network` would be sent in the response with whatever the value of the `Access-Control-Request-Private-Network` was.  That maybe creates the potential for strange bugs so maybe just having it be `"true"` might be better?
- If `CORS_ALLOW_PRIVATE_NETWORK_ACCESS` is set to `"true"`, the `Access-Control-Allow-Private-Network` would be sent in the response with the value `"false"`
- If `CORS_ALLOW_PRIVATE_NETWORK_ACCESS` is not specified, no header will be sent `add_header` won't send the header if given a falsey value).

It would be possible switch on the presence of the header and its value by using a `map` directive in the `default.conf` file but I'm a little hesitant to go there because it introduces a lot of complexity for a specific use case that is still in the proposal stage.

Interested to hear your thoughts.

Reply:
> It would be possible switch on the presence of the header and its value by using a map directive in the default.conf file but I'm a little hesitant to go there because it introduces a lot of complexity for a specific use case that is still in the proposal stage.

Yeah, I agree that it's not worth burning too much complexity for this.

> We could implement your proposed CORS_ALLOW_PRIVATE_NETWORK_ACCESS as I suggested in my comment (always returning true if enabled). But without any awareness of the actual headers received.

I think this is a good option. From my reading of the spec, there should be no issues with returning this header all the time. The [spec](https://wicg.github.io/private-network-access/) only contemplates a single value of true being checked for as the successful response value.

Do you want me to update the PR? I'm happy for you to if you'd like as well.

Reply:
Yeah if you don't mind can you please update the PR and verify that the code still works for your use case that would be awesome.

The test suite is a little harder to update so I'd be happy to add tests for you. If you [allow me to push](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) I can do that once your updates are in.